src/inmanta/module.py

```diff
@@ -1104,6 +1241,20 @@ def get_metadata_file_schema_type(cls) -> Type[T]:
     def get_name_from_metadata(cls, metadata: T) -> str:
         raise NotImplementedError()
 
+    @abstractmethod
+    def add_module_requirement(self, requirement: InmantaModuleRequirement, add_as_v1_module: bool) -> None:
```

Comment:
I would very clearly document that his is about re-writing the on-disk config

(perhaps even add it to the name in some way? like `add_module_requirement_persistent` )